core/templates/pages/skill-editor-page/editor-tab/skill-concept-card-editor/worked-example-editor.component.spec.ts

```diff
@@ -0,0 +1,160 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for WorkedExampleEditorComponent
+ */
+
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { ChangeDetectorRef, NO_ERRORS_SCHEMA } from '@angular/core';
+import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
+import { SkillUpdateService } from 'domain/skill/skill-update.service';
+import { WorkedExampleEditorComponent } from './worked-example-editor.component';
+
+describe('Worked example editor component', function() {
+  let component: WorkedExampleEditorComponent;
+  let fixture: ComponentFixture<WorkedExampleEditorComponent>;
+  let skillUpdateService: SkillUpdateService;
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      declarations: [
+        WorkedExampleEditorComponent
+      ],
+      providers: [
+        ChangeDetectorRef,
+        SkillUpdateService
+      ],
+      schemas: [NO_ERRORS_SCHEMA]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(WorkedExampleEditorComponent);
+    component = fixture.componentInstance;
+    skillUpdateService = TestBed.inject(SkillUpdateService);
+
+    component.isEditable = true;
+    component.index = 2;
+    component.workedExample = {
+      getQuestion(): object {
+        return {
+          html: 'worked example question 1',
+          content_id: 'worked_example_q_1',
+        };
+      },
+
+      getExplanation(): object {
+        return {
+          html: 'worked example explanation 1',
+          content_id: 'worked_example_e_1',
+        };
+      }
+    };
+    component.ngOnInit();
+  });
+
+  it('should set properties when initialized', () => {
+    expect(component.questionEditorIsOpen).toBe(false);
+    expect(component.explanationEditorIsOpen).toBe(false);
+    expect(component.WORKED_EXAMPLE_FORM_SCHEMA).toEqual({
+      type: 'html',
+      ui_config: {}
+    });
+  });
+
+  it('should open question editor when calling ' +
+    '\'openQuestionEditor\'', () => {
+    expect(component.questionEditorIsOpen).toBe(false);
+
+    component.openQuestionEditor();
+
+    expect(component.questionEditorIsOpen).toBe(true);
+  });
+
+  it('should open explanation editor when calling ' +
```

Comment:
Please write the `it` block description from the user's perspective.

Reply:
Done.